File: python-sdk-examples/completion/chat_stream.py
from writerai import Writer
import dotenv

dotenv.load_dotenv()

client = Writer()

def main():
    try:
        chat = client.chat.chat(
            messages=[
                {
                    "role": "user",
                    "content": "You are an expert at writing concise product descriptions for an E-Commerce Retailer"
                },
                {
                    "role": "assistant",
                    "content": "Okay, great I can help write these descriptions. Do you have a specific product in mind?"
                },
                {
                    "role": "user",
                    "content": "Please write a one sentence product description for a cozy, stylish sweater suitable for both casual and formal occasions"
                }
            ],
            model="palmyra-x-004",
            stream=True,
            logprobs=True,
            stream_options={
                "include_usage": True
            }
        )
        
        response_text=""
        for chunk in chat:
            if len(chunk.choices) > 0 and chunk.choices[0].delta.content is not None:
                response_text += chunk.choices[0].delta.content
        
        print(response_text)
    except Exception as error:
        print("Error:", error)

if __name__ == "__main__":
    main()

File: node-sdk-examples/completion/chat-stream.js
const Writer = require('writer-sdk');
const dotenv = require('dotenv');

dotenv.config();

const client = new Writer({
    apiKey: process.env.WRITER_API_KEY,
});

async function main() {
    try {
      const chat = await client.chat.chat({
        messages: [
          {
              role: "user",
              content: "You are an expert at writing concise product descriptions for an E-Commerce Retailer"
          },
          {
              role: "assistant",
              content: "Okay, great I can help write these descriptions. Do you have a specific product in mind?"
          },
          {
              role: "user",
              content: "Please write a one sentence product description for a cozy, stylish sweater suitable for both casual and formal occasions"
          }
        ],
        model: 'palmyra-x-004',
        stream: true
      });
      

      console.log('Streaming response:');
      let content = '';
      for await (const chunk of chat) {
        if (chunk.choices[0].delta.content) {
          content += chunk.choices[0].delta.content;
        }
      }
      console.log(content);
    } catch (error) {
      console.error('Error:', error);
    }
  }

main();